ray-operator/controllers/ray/rayjob_controller_scheduled_test.go

```diff
@@ -0,0 +1,222 @@
+				time.Second*3, time.Millisecond*500).Should(Succeed())
+		})
+
+		// We are checking if the LastScheduleTime is correctly set
+		It("should have LastScheduleTime updated in its status", func() {
+			rayJobLookupKey := types.NamespacedName{Name: rayJob.Name, Namespace: rayJob.Namespace}
+			fetchedRayJob := &rayv1.RayJob{}
+
+			var lastScheduleTime *time.Time
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, rayJobLookupKey, fetchedRayJob)
+				if err != nil {
+					return false
+				}
+				if fetchedRayJob.Status.LastScheduleTime != nil {
+					lastScheduleTime = &fetchedRayJob.Status.LastScheduleTime.Time
+					return true
+				}
+				return false
+			}, time.Second*10, time.Millisecond*500).Should(BeTrue(), "expected LastScheduleTime to be set")
+
+			Expect(lastScheduleTime).ToNot(BeNil(), "LastScheduleTime should not be nil")
+			Expect(*lastScheduleTime).ToNot(BeZero(), "LastScheduleTime should not be a zero time")
+
+			Expect(lastScheduleTime.After(time.Now().Add(-15*time.Second))).To(BeTrue(), "LastScheduleTime should be within the last 15 seconds")
+			Expect(lastScheduleTime.Before(time.Now().Add(5*time.Second))).To(BeTrue(), "LastScheduleTime should not be in the future")
+
+			GinkgoWriter.Printf("Validated LastScheduleTime: %s\n", lastScheduleTime.String())
+		})
+
+		It("should NOT create the underlying K8s job yet because the cluster is not ready", func() {
+			underlyingK8sJob := &batchv1.Job{}
+			Consistently(
+				// k8sClient client throws error if resource not found
+				func() bool {
+					err := getResourceFunc(ctx, common.RayJobK8sJobNamespacedName(rayJob), underlyingK8sJob)()
+					return errors.IsNotFound(err)
+				},
+				time.Second*3, time.Millisecond*500).Should(BeTrue())
+		})
+
+		It("should be able to update all Pods to Running", func() {
+			updateHeadPodToRunningAndReady(ctx, rayJob.Status.RayClusterName, namespace)
+			updateWorkerPodsToRunningAndReady(ctx, rayJob.Status.RayClusterName, namespace)
+		})
+
+		It("Dashboard URL should be set", func() {
+			Eventually(
+				getDashboardURLForRayJob(ctx, rayJob),
+				time.Second*3, time.Millisecond*500).Should(HavePrefix(rayJob.Name), "Dashboard URL = %v", rayJob.Status.DashboardURL)
+		})
+
+		It("should create the underlying Kubernetes Job object", func() {
+			underlyingK8sJob := &batchv1.Job{}
+			// The underlying Kubernetes Job should be created when the RayJob is created
+			Eventually(
+				getResourceFunc(ctx, common.RayJobK8sJobNamespacedName(rayJob), underlyingK8sJob),
+				time.Second*3, time.Millisecond*500).Should(Succeed(), "Expected Kubernetes job to be present")
+		})
+
+		It("should transition to the Running", func() {
+			Eventually(
+				getRayJobDeploymentStatus(ctx, rayJob),
+				time.Second*5, time.Millisecond*500).Should(Equal(rayv1.JobDeploymentStatusRunning),
+				"JobDeploymentStatus should be Running")
+		})
+
+		It("RayJobs's JobDeploymentStatus transitions to Scheduled after Job is Complete.", func() {
```

Comment:
The RayJob's jobDeploymentStatus transitioning to complete should be it's own IT block like this one https://github.com/ray-project/kuberay/blob/master/ray-operator/controllers/ray/rayjob_controller_test.go#L496-L526

And then there should be a separate IT block right after validating the transition to Scheduled